inc/saiport.h

```diff
@@ -431,6 +431,15 @@ typedef enum _sai_port_attr_t
      * given port list will be dropped. */
     SAI_PORT_ATTR_EGRESS_BLOCK_PORT_LIST,
 
+    /** Port Hardware Configuration Profile ID [sai_uint64_t]
+     * Port can require different hardware configuration based on the attached
+     * media type, cable length etc. A Profile ID maps to a Port Hardware
+     * configuration settings that needs to be applied on the Port.
+     * This attribute need not be implemented and can be ignored if the port
+     * doesn't require any specific hardware settings based on media type/cable.
+     */
+    SAI_PORT_ATTR_HW_PROFILE_ID,
```

Comment:
Hi @atitjain 
Your understanding is correct except that it's not intended to mean a collection of certain switch configurations (as you mentioned in point 1), instead it's a collection of certain port specific hardware configuration.
Example could be to configure the port interface type to SR4, speed to 40G and any special phy tuning paramters when QSFP SR4 type optic is inserted.